.github/workflows/deploy-gh-pages.yml

```
@@ -0,0 +1,42 @@
+name: "deploy-github-pages"
+
+on:
+  push:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - uses: "actions/checkout@v4"
+
+      - uses: "actions/setup-node@v4"
+        with:
+          node-version: "22"
+          cache: "npm"
+          cache-dependency-path: "./package-lock.json"
+
+      - run: "npm install"
+      - run: "npm run build"
+
+      - name: "Upload static files as artifact"
+        uses: "actions/upload-pages-artifact@v3"
+        with:
+          path: "./dist"
+
+  deploy:
+    needs: "build"
+    permissions:
+      # to deploy to Pages
+      pages: "write"
+      # to verify the deployment originates from an app
```

```suggestion

      # To request a JWT from GitHub
```

Based on https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings

---

How about
```suggestion

      # To request a JWT from GitHub to verify the deployment originates from an appropriate source
```
I was referring to this: https://github.com/actions/deploy-pages#:~:text=deploy%20to%20Pages-,id%2Dtoken,-%3A%20write%20%20%20%23 (though I didn't notice the code block was truncated in my browser... sorry

---

This is where the token is used: https://docs.github.com/en/rest/pages/pages?apiVersion=2022-11-28#:~:text=Default%3A%20GITHUB_SHA-,oidc_token,-string%20Required

---

Makes sense.